### PR TITLE
Compute a possible estimate of the percent of each ingredients (with the sum equal to 100%)

### PR DESCRIPTION
**Pull request by stephanegigandet**
See #4423 

This is needed in particular for the Eco-Score, so that we can weight each ingredient.

### CODE REVIEW

**Comment by svensven** [comment]
Doesn't this happen already? Or is this a replacement for that code?

On this topic, it might be nice if the system could understand all the american lists with "contains 2% or less of the following" type text in them, when calculating percentages. (or maybe it already does, but just doesn't take it out of the unknown ingredients.)

**Comment by stephanegigandet** [comment]
> Doesn't this happen already? Or is this a replacement for that code?

It is an extension of the existing code:

- we have been computing the possible percent min and max for all ingredients and sub-ingredients (based on actual % provided and the position of each ingredient in the list). but this only gives us possible ranges for each ingredient.

- for some uses, we need an estimate (not a range), and we need the sum of all those estimates to be 100%. This is what this new code does: computing those "estimates" (that at this point are more a set of possible values that match all the min and max constraints).

**Comment by stephanegigandet** [comment]
> On this topic, it might be nice if the system could understand all the american lists with "contains 2% or less of the following" type text in them, when calculating percentages. (or maybe it already does, but just doesn't take it out of the unknown ingredients.)

It definitely would be nice, but we don't have code for that yet.

File: lib/ProductOpener/Ingredients.pm [hunk 1]
```diff
@@ -1976,6 +1979,98 @@ sub set_percent_sub_ingredients($) {
 }
 
 
+=head2 compute_ingredients_percent_values ( total_min, total_max, ingredients_ref )
```
Comment (svensven) [inline lib/ProductOpener/Ingredients.pm:1982]: this looks like the blurb for the wrong function?

Comment (stephanegigandet) [reply lib/ProductOpener/Ingredients.pm]: indeed, I just changed it.

**Review by svensven — APPROVED** [review]

**Pull request merged**